context/src/main/java/io/micronaut/scheduling/executor/GracefulShutdownCapableScheduledThreadPoolExecutor.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.scheduling.executor;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.runtime.server.GracefulShutdownCapable;
+
+import java.util.OptionalLong;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Extension of {@link java.util.concurrent.Executors#newScheduledThreadPool(int, ThreadFactory)}
+ * that supports graceful shutdown.
+ *
+ * @since 4.9.0
+ * @author Jonas Konrad
+ */
+@Internal
+final class GracefulShutdownCapableScheduledThreadPoolExecutor extends ScheduledThreadPoolExecutor implements GracefulShutdownCapable {
```

Comment:
should this only be a activated if graceful shutdown is enabled?

Reply:
So far, all the GracefulShutdownCapable beans are enabled unconditionally. In fact, you can use GracefulShutdownManager without the config option. Only the ShutdownEvent listener is gated.

I think the overhead is negligible, and it allows for more flexibility for programmatic use for those that don't want to use the listener